internal/downloader/downloader.go

```diff
@@ -53,6 +57,23 @@ func Download(ctx context.Context, destDir string, sourceUrl string, showMsg boo
 		return fmt.Errorf("attempting to download from insecure source: %s", sourceUrl)
 	}
 
+	// read the USEGOGATHER env variable to determine which downloader to use:
+	// - if USEGOGATHER=0, use the Conftest downloader
+	// - if USEGOGATHER=1, use the go-gather downloader
+	// - if USEGOGATHER is not set, use the Conftest downloader
+	// - if USEGOGATHER is set to any other value, use the Conftest downloader
+	// This is a temporary solution to allow for testing the go-gather downloader
+	// without breaking the existing functionality.
+	// Once the go-gather downloader is stable, the USEGOGATHER env variable
+	// will be removed and the go-gather downloader will be used by default.
+	// The USEGOGATHER env variable will be removed in a future release.
+	if experimental := os.Getenv("USEGOGATHER"); experimental != "" {
```

Comment:
Perhaps it is not necessary to read the environment variable twice? Perhaps something like:

```
	dl := func(ctx context.Context, dest, src string) error {
		// conftest's Download function leverages oras under the hood to fetch from OCI. It uses the
		// global oras client and sets the user agent to "conftest". This is not a thread safe
		// operation. Here we get around this limitation by ensuring a single download happens at a
		// time.
		dlMutex.Lock()
		defer dlMutex.Unlock()
		return downloader.Download(ctx, dest, []string{src})
	}
	if os.Getenv("USEGOGATHER") == "1" {
		dl = func(ctx context.Context, dest, src string) error {
			_, err := gather.Gather(ctx, src, dest)
			return err
		}
	} else if d, ok := ctx.Value(downloadImplKey).(downloadImpl); ok {
		dl = func(ctx context.Context, dest, src string) error {
			return d.Download(ctx, dest, []string{src})
		}
	}

	err = dl(ctx, destDir, []string{sourceUrl})

Reply:
Not sure, but we might want to use the mocked downloader for unit tests even when USEGOGATHER is set. So I think the `if d, ok := ctx.Value(downloadImplKey).(downloadImpl); ok` part should be the first check instead of the last.

In pseudo code:
```
  if downloader impl found in context:
    // must be in unit test, downloads are mocked probably
  else if go gather env flag is set:
    // use go gather
  else:
    // good old conftest download
  fi
```

Reply:
Being that the intention is to move to `go-gather` in lieu of the `conftest-downloader`, I'm not sure this is really worth much in the current time frame. We can update this when / if needed when we transition to using `go-gather` exclusively. At that time we'll do away with looking at the `USEGOGATHER` env var anyway.